### PR TITLE
2015-day-21-part-1

### PR DESCRIPTION
**Pull request by Herrera93**
Created a library to find the most cost effective items in a turn based RPG.

### CODE REVIEW

**Comment by Herrera93** [comment]
[2015-day-21-part-1](https://trello.com/c/i4HDfPwC/24-2015-day-21-part-1)

File: src/year2015/day_21.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    pub name: String,
+    pub cost: i32,
+    pub damage: i32,
+    pub defense: i32,
+}
+
+impl FromStr for Item {
+    type Err = ItemError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```
Comment (kanekv) [inline src/year2015/day_21.rs]: next time this could be done without parser.

Comment (Herrera93) [reply src/year2015/day_21.rs]: How would it be better ? 
Adding the Item without parsing it ? Like
```
market.add_weapon(Item { 
     name: "Dagger".to_string(),
     cost: 8,
     damage: 4,
     defense: 0
})
```

Comment (kanekv) [reply src/year2015/day_21.rs]: @Herrera93 yeah, in this case that would work I think, as iterms are predefined and there is not too many of them. What you did is good too.

**Pull request merged**